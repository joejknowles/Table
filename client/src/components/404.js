import React from 'react';
import res from '../resources/pages/404';

export default () => (
  <div className="not-found-message">
    { res.message }
  </div>
);
