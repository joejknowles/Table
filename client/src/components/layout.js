import React from 'react';

import Notifications from './notifications';

export default ({ children }) => (
  <div className="App">
    { children }
    <Notifications />
  </div>
);
