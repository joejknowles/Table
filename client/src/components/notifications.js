import React from 'react';
import { connect } from 'react-redux';
import { notificationSelector } from '../reducers';

export const Notifications = ({ notification }) => (
  notification ? <p>{ notification }</p> : null
);

const mapStateToProps = (state) => ({
  notification: notificationSelector(state)
});

export default connect(mapStateToProps)(Notifications);
