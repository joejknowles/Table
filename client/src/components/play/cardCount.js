import React from 'react';
import { connect } from 'react-redux';

import res from '../../resources/pages/play';

import { cardCountSelector } from '../../reducers';

export const CardCount = ({ count }) => (
  <div className="card-count">
    { res.cardCount(count) }
  </div>
);

const mapStateToProps = (state) => ({
  count: cardCountSelector(state)
});

export default connect(mapStateToProps)(CardCount);
