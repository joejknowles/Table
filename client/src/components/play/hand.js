import React from 'react';

import PlayCardButton from './playCardButton';

export default () => (
  <div className='card'>
    <PlayCardButton />
  </div>
);
