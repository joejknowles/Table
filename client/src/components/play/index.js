import React from 'react';
import { connect } from 'react-redux';

import Hand from './hand';
import { handSelector } from '../../reducers';
import '../../styles/Play.css';
import res from '../../resources/pages/play';

export const Play = ({ hand }) => (
  <div className="App Play">
    <div className="deck">
      { hand > 0 ?
          <Hand /> :
          <p className='no-cards-message'>
            { res.noCardsMessage }
          </p>
      }
    </div>
  </div>
);


export default connect(
  (state) => ({ hand: handSelector(state) })
)(Play);
