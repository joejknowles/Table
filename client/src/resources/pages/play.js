export default {
  playCard: 'Play Card',
  noCardsMessage: 'No more cards'
};
