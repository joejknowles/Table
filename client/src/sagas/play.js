import { call, takeEvery } from 'redux-saga/effects';

import * as toPath from '../routing';
import * as events from '../api/sockets';

export function* playCard(socket, action) {
  yield call(events.playCard, socket, action.card)
}

export function* playerBegin(socket) {
  yield takeEvery('PLAY_CARD', playCard, socket);
  yield call(toPath.play);
}

export function* playerJoin(socket) {
  yield call(events.joinPlayersRoom, socket);
  yield takeEvery('BEGIN_GAME', playerBegin, socket);
  yield call(toPath.waiting);
}
