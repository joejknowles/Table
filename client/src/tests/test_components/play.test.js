import { Play } from '../../components/play';
import React from 'react';
import ReactDOM from 'react-dom';

import ReactTestUtils from 'react-addons-test-utils';
const shallowRenderer = ReactTestUtils.createRenderer();


describe('Play page', () => {
  it('renders as before with cards', () => {
    const component = shallowRenderer.render(<Play cardsRemaining={ 1 } />);
    expect(component).toMatchSnapshot();
  });

  it('renders as before without cards', () => {
    const component = shallowRenderer.render(<Play cardsRemaining= { 0 } />);
    expect(component).toMatchSnapshot();
  });
});
