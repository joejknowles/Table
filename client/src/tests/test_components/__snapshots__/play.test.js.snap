exports[`test component renders as before 1`] = `
<div
  className="App Play">
  <div
    className="deck">
    <Unknown />
  </div>
</div>
`;
